### PR TITLE
POC: load dashboard json from a plugin

### PR DESCRIPTION
**Pull request by ryantxu**
There have been many recent discussions about dynamic data-driven dashboards.  In particular, @sdboyer is looking at how to evolve mixins into something more.  

I think we need to expose some method for plugins to develop dynamic dashboards ASAP, and figure out how this will integrate into a unified search later.

This POC adds a new dashboard type 'plugin' that passes the rest of the URL path to a plugin that can then load dashboard json.  This is currently a frontend plugin with a hidden interface -- long term, I think this should be in the gRPC contract and managed in the backend plugin... BUT we should not block progress waiting for the more complicated path :)

In this PR, the URL:
http://localhost:3000/dashboard/p/dashgen/path/to/whatever?orgId=1

asks the `dashgen` plugin to load a dashboard for `path/to/whatever` -- in this case a super simple hardcoded dashboard:

image

### CODE REVIEW

File: public/app/plugins/panel/dashgen/loader.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import { DashboardLoaderSupport } from 'app/features/dashboard/types';
+import { DashboardDTO } from 'app/types';
+import { of } from 'rxjs';
+
+export const loader: DashboardLoaderSupport = {
+  loadDashboard: (path: string) => {
+    console.log('LOAD', path);
+    return of(superSimple as DashboardDTO);
```
Comment (ryantxu) [inline public/app/plugins/panel/dashgen/loader.ts:8]: @sdboyer -- this example just loads a static definition, but if you have a simple HTTP server running we can load that dynamically

**Comment by sdboyer** [comment]
Brilliant, thank you for taking first steps here! I definitely agree that standardizing on some approaches to programatically generated dashboards has enormous potential.

> and figure out how this will integrate into a unified search later.

For my edification - when you say "unified search," you're referring [dashboard search](https://grafana.com/docs/grafana/latest/dashboards/search/), correct?

**Comment by ryantxu** [comment]
> when you say "unified search," you're referring dashboard search, correct?

yes.  If we have an extendable way to generate dashboards, I think we still need them in the search results

**Comment by ryantxu** [comment]
Just a POC... will close for now

**Pull request closed without merging**